apps/api/src/modules/payments/payments.service.ts

```diff
@@ -745,7 +745,7 @@ export default class PaymentsService {
       pendingPayments.map(async (payment) => {
         let status: PaymentStatus | undefined
         // Card flow
-        if (payment.externalId && payment.paymentCardId) {
+        if (payment.externalId) {
```

Comment:
Checking for payment.paymentCardId here causes cards a user did not select to save for later use to never have their payment status updated. (In production every payment comes in as pending, even if it can be updated a few seconds later).